server/src/internalClusterTest/java/org/opensearch/snapshots/SegmentReplicationSnapshotIT.java

```diff
@@ -0,0 +1,250 @@
+
+    public RestoreSnapshotResponse restoreSnapshotWithSettings(Settings indexSettings) {
+        RestoreSnapshotRequestBuilder builder = client().admin()
+            .cluster()
+            .prepareRestoreSnapshot(REPOSITORY_NAME, SNAPSHOT_NAME)
+            .setWaitForCompletion(false)
+            .setRenamePattern(INDEX_NAME)
+            .setRenameReplacement(RESTORED_INDEX_NAME);
+        if (indexSettings != null) {
+            builder.setIndexSettings(indexSettings);
+        }
+        return builder.get();
+    }
+
+    public void testRestoreOnSegRep() throws Exception {
+        startClusterWithSettings(segRepEnableIndexSettings());
+        createSnapshot();
+        // Delete index
+        client().admin().indices().delete(new DeleteIndexRequest(INDEX_NAME));
+        Thread.sleep(5000);
+        assertFalse("index [" + INDEX_NAME + "] should have been deleted", indexExists(INDEX_NAME));
+
+        logger.info("Restore from snapshot");
+        RestoreSnapshotResponse restoreSnapshotResponse = restoreSnapshotWithSettings(null);
+
+        // Assertions
+        assertThat(restoreSnapshotResponse.status(), equalTo(RestStatus.ACCEPTED));
+        logger.info("Ensure cluster is green");
+        ensureGreen(RESTORED_INDEX_NAME);
+        GetSettingsResponse settingsResponse = client().admin()
+            .indices()
+            .getSettings(new GetSettingsRequest().indices(RESTORED_INDEX_NAME))
+            .get();
+        assertEquals(settingsResponse.getSetting(RESTORED_INDEX_NAME, "index.replication.type"), "SEGMENT");
+        SearchResponse resp = client().prepareSearch(RESTORED_INDEX_NAME).setQuery(QueryBuilders.matchAllQuery()).get();
+        assertHitCount(resp, DOC_COUNT);
+    }
+
+    public void testRestoreOnSegRepDuringIngestion() throws Exception {
+        startClusterWithSettings(segRepEnableIndexSettings());
+        createSnapshot();
+        // Delete index
+        client().admin().indices().delete(new DeleteIndexRequest(INDEX_NAME));
+        Thread.sleep(5000);
+        assertFalse("index [" + INDEX_NAME + "] should have been deleted", indexExists(INDEX_NAME));
+
+        logger.info("Restore from snapshot");
+        RestoreSnapshotResponse restoreSnapshotResponse = restoreSnapshotWithSettings(null);
+
+        // Assertions
+        assertThat(restoreSnapshotResponse.status(), equalTo(RestStatus.ACCEPTED));
+        logger.info("Ensure cluster is green");
+        ingestData(5000, RESTORED_INDEX_NAME);
+        ensureGreen(RESTORED_INDEX_NAME);
+        GetSettingsResponse settingsResponse = client().admin()
+            .indices()
+            .getSettings(new GetSettingsRequest().indices(RESTORED_INDEX_NAME))
+            .get();
+        assertEquals(settingsResponse.getSetting(RESTORED_INDEX_NAME, "index.replication.type"), "SEGMENT");
+        SearchResponse resp = client().prepareSearch(RESTORED_INDEX_NAME).setQuery(QueryBuilders.matchAllQuery()).get();
+        assertHitCount(resp, DOC_COUNT);
```

Comment:
This looks to me like it should fail? You are ingesting 5k docs and asserting there are 1010?

Reply:
Thanks @mch2 for catching this. The `ingestData` method has issue where `INDEX_NAME` index was hard-coded for indexing and `indexName` passed param was ignored (was only used for refresh).

https://github.com/opensearch-project/OpenSearch/pull/3993/files#diff-4ca892b67e6bb252096508fccda6dcaa9b3acdf1c24b4b42f47a3d25b19deddbR73